core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -2482,6 +2482,41 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
     assertThat(inputRef1.getIdentifier(), is(inputRef2.getIdentifier()));
   }
 
+  @Test void testExpressionLineageConjuntiveExpression() {
+    // empno is column 0 in catalog.sales.emp
+    // ename is column 1 in catalog.sales.emp
+    // deptno is column 7 in catalog.sales.emp
+    final RelNode rel = convertSql("select (empno = 1 or ename = 'abc') and deptno > 1 from emp");
```

Comment:
It would be great if you could move those comments and the SQL in question to the last `assertEquals` in the test.

Then the assertion call would be paired with the intention, and the failure message would include the intended behavior as well.

Reply:
I have added an assertion message, failure now look like this:
```
For query 'select (empno = 1 or ename = 'abc') and deptno > 1 from emp':
'empno' is column 0 in 'catalog.sales.emp', 'ename' is column 1 in 'catalog.sales.emp', and 'deptno' is column 7 in 'catalog.sales.emp' ==> expected: <AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1)> but was: <AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1))>
Expected :AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1)
Actual   :AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1))
```